packages/web3-core-method/src/index.js

```diff
@@ -393,8 +393,12 @@ Method.prototype._confirmTransaction = function(defer, result, payload) {
                 // CHECK for normal tx check for receipt only
                 .then(function(receipt) {
                     if (!isContractDeployment && !promiseResolved) {
+
+                        var hasStatus = receipt.status === false || receipt.status === true;
+                        var isOOG = !hasStatus && gasProvided === utils.numberToHex(receipt.gasUsed);
+
                         if (!receipt.outOfGas &&
-                            (!gasProvided || gasProvided !== utils.numberToHex(receipt.gasUsed)) &&
+                            (!gasProvided || !isOOG) &&
                             (receipt.status === true || receipt.status === '0x1' || typeof receipt.status === 'undefined')) {
```

Comment:
- What exactly are the thoughts behind ``hasStatus``? (Because the ``status`` field does always exists in post Byzantium hard fork chains)
- The ``if`` conditions who check the status based on the hex value e.g.: ``receipt.status === '0x1'`` aren't required because the ``receipt.status`` property does get type casted to a boolean value type in the ``outputTransactionReceiptFormatter`` of the used ``getTransactionReceipt`` method.
  + If the status property is of type ``undefined`` or ``null`` is no type cast applied.
- Shouldn't ``isOOG`` be like this: ``isOOG = !receipt.status && gasProvided === utils.numberToHex(receipt.gasUsed);`` ?
- Shouldn't we add [here](https://github.com/ethereum/web3.js/pull/3189/files#diff-b87e4847acaf5b45ad88f78ed46732dbR414) the ``isOOG`` condition as well? 
  + I would propose those conditions: 
    ``` javascript 
      if(receipt.status === true || receipt.status === null || typeof receipt.status === 'undefined' ) {
        // resolve with receipt
      } else {
        if (gasProvided === utils.numberToHex(receipt.gasUsed)) {
          // reject with out of gas
        } else {
          // reject with reverted by EVM
        }
      }
    ```

Reply:
> What exactly are the thoughts behind hasStatus

My thought is that the reason for the original correction in #3123 is to support non-byzantium chains where gasUsed === gasSent is (almost) logically equivalent to status === false. What is your view of this? 

> '0x1' aren't required 

👍 Will remove

> Shouldn't isOOG be like this: `isOOG = !receipt.status...`

That was my initial thought as well. But it's a problem when people combine `estimateGas` with tx's that actually revert. In that case receipt status is false *and* exact gas is consumed. See [this added test][1]. My theory is that preferring the status / ignoring the gas is closest to what the 1.2.1 behavior is.   

If you look at what GabMontes [suggests][2] in #3175, there are other receipt fields they're looking at to disambiguate real OOG from revert-like transaction failure. I'm ambivalent about adding those without being able to add a test against an ETC client here. 
[1]: https://github.com/ethereum/web3.js/blob/b24dfd5c32c28d9a7863eb7f5c3bca23603a999c/test/e2e.method.send.js#L83-L104
[2]: https://github.com/ethereum/web3.js/issues/3175#issuecomment-549385405

Reply:
Ah! I just realized the test has a copy-pasta mistake in it. Let me investigate this further...